### PR TITLE
[Resource] Bring back a default form type

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | partially #7041 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/Doctrine/DoctrineORMDriver.php [hunk 1]
```diff
@@ -25,6 +27,15 @@
  */
 final class DoctrineORMDriver extends AbstractDoctrineDriver
 {
+    public function load(ContainerBuilder $container, MetadataInterface $metadata)
```
Comment (Zales0123) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/Doctrine/DoctrineORMDriver.php]: Where is the docblock? :(

File: src/Sylius/Bundle/ResourceBundle/test/src/AppBundle/Entity/ComicBook.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    private $author;
+
+    /**
+     * @var string
+     */
+    private $title;
+
+    /**
+     * @return int
+     */
```
Comment (Zales0123) [inline src/Sylius/Bundle/ResourceBundle/test/src/AppBundle/Entity/ComicBook.php]: `{@inheritdoc}`

File: src/Sylius/Bundle/ResourceBundle/test/src/Tests/Controller/ComicBookApiTest.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+        $this->client->request('GET', '/comic-books/');
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'comic-books/index_response');
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_showing_resource_if_it_not_exists()
```
Comment (Zales0123) [inline src/Sylius/Bundle/ResourceBundle/test/src/Tests/Controller/ComicBookApiTest.php]: `it_does_not_allow_showing_resource_if_it_does_not_exist`

**Comment by lchrusciel** [comment]
@pjedrzejewski It works fine with more then one form defined, but you are right. This form type should be declared only once. On it. 💨

File: src/Sylius/Bundle/ResourceBundle/Resources/config/services/integrations/doctrine/orm.xml [hunk 4]
```diff
@@ -26,5 +26,13 @@
             <argument type="service" id="sylius.resource_registry" />
             <tag name="doctrine.event_subscriber" priority="8192" />
         </service>
+        <service id="sylius.form_builder.default" class="Sylius\Bundle\ResourceBundle\Doctrine\ORM\Form\Builder\DefaultFormBuilder" public="false">
+            <argument type="service" id="doctrine.orm.default_entity_manager" />
+        </service>
+        <service id="sylius.form.type.default" class="Sylius\Bundle\ResourceBundle\Form\Type\DefaultResourceType">
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Resources/config/services/integrations/doctrine/orm.xml]: This service is not related to ORM.

File: src/Sylius/Bundle/ResourceBundle/Form/Type/DefaultResourceType.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+final class DefaultResourceType extends AbstractType
+{
+    /**
+     * @var RegistryInterface
+     */
+    private $resourceRegistry;
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Form/Type/DefaultResourceType.php]: It's rather *resource metadata registry* or *metadata registry*.

File: src/Sylius/Bundle/ResourceBundle/Resources/config/services/integrations/doctrine/orm.xml [hunk 6]
```diff
@@ -26,5 +26,9 @@
             <argument type="service" id="sylius.resource_registry" />
             <tag name="doctrine.event_subscriber" priority="8192" />
         </service>
+        <service id="sylius.form_builder.default" class="Sylius\Bundle\ResourceBundle\Doctrine\ORM\Form\Builder\DefaultFormBuilder" public="false">
+            <argument type="service" id="doctrine.orm.default_entity_manager" />
+            <tag name="sylius.form_builder" type="doctrine/orm"/>
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Resources/config/services/integrations/doctrine/orm.xml]: Missing space between `"` and `/>`. :taco:

**Comment by lchrusciel** [comment]
I have changed it to 'sylius.default_resource_form.builder' to be as accurate as possible. If anyone has better idea, please, share it :)

**Comment by pjedrzejewski** [comment]
Thank you Łukasz!

**Comment by michalmarcinkowski** [comment]
👍 👍
